api-flask/app/main.py

```diff
@@ -103,41 +105,73 @@ def stats():
     return jsonify(features)
 
 
-@app.route('/alerts', methods=['POST'])
+@app.route('/alerts-all', methods=['GET'])
+def alerts_all():
+    results = alert_db.readall()
+    return Response(json.dumps(results, cls=AlchemyEncoder), mimetype='application/json')
+
+
+@app.route('/alerts', methods=['POST', 'GET'])
 def alerts():
-    """Post new alerts."""
     if request.method == 'POST':
-        if not request.is_json:
-            logger.error('Unrecognized operation. JSON data expected.')
-            return Response(
-                response='500: Unrecognized operation. JSON data expected',
-                status=500
-            )
-
-        data = json.loads(request.get_data())
-        alert = AlertModel(**data)
-        try:
-            alert_db.write(alert)
-            return Response(response='Success', status=200)
-        except Exception as e:
-            logger.error(e)
-            return Response(
-                response='500: OperationalError.',
-                status=500
-            )
+        return _write_alert(request)
+    elif request.method == 'GET':
+        return _read_alert_id(request)
+    else:
+        return Response(
+            response='500: OperationalError',
+            status=500
+        )
+
+
+def _write_alert(request):
+    if not request.is_json:
+        return Response(
+            response='500: InvalidInput',
+            status=500
+        )
+    data = json.loads(request.get_data())
+    data['created_at'] = datetime.datetime.now()  # if don't provide timestamp, it will
```

Comment:
the DB will handle this automatically

Reply:
When I tested locally, this field is always the same for all test data, it's using the timestamp when DB initialized.

Reply:
That's because when you set the default in the ORM definition you executed the function ()

So the default is the date of when your ran the script the first time instead of now() when the row is created

Reply:
Make sense, i should pass a function instead of a value.